src/servicesHost.ts

```diff
@@ -134,6 +130,42 @@ function makeResolversAndModuleResolutionHost(
   }
 }
 
+function makeResolversAndModuleResolutionHost(
+  scriptRegex: RegExp,
+  loader: webpack.LoaderContext<LoaderOptions>,
+  instance: TSInstance,
+  fileExists: (fileName: string) => boolean,
+  enableFileCaching: boolean
+) {
+  const {
+    compiler,
+    compilerOptions,
+    appendTsTsxSuffixesIfRequired,
+    loaderOptions: {
+      resolveModuleName: customResolveModuleName,
+      resolveTypeReferenceDirective: customResolveTypeReferenceDirective,
+    },
+  } = instance;
+
+  
+
+  // make a (sync) resolver that follows webpack's rules
+  const resolveSync = makeResolver(loader._compiler!.options);
```

Comment:
Do you really need this? Based on logic it is just an error - https://github.com/TypeStrong/ts-loader/blob/main/src/resolver.ts#L11

Webpack has great API for resolving, you can use (just an example of code, we should have two resolvers for cjs and esm in the real life):
```
const resolve = loaderContext.getResolve({
    dependencyType: "typescript",
    conditionNames: ["types", "..."],
    mainFiles: ["index", "..."],
    extensions: [".ts", ".cts", "..."],
});
```

And hide a lot of options for basic `resolve` under the hood - less options better DX, but yeah, I think it should be done in another PR, here is just about ESM

Reply:
This is just moved code, not code I wrote